pkg/asset/machines/machineconfig/ipv6.go

```diff
@@ -0,0 +1,42 @@
+package machineconfig
+
+import (
+	"fmt"
+
+	igntypes "github.com/coreos/ignition/v2/config/v3_2/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/installer/pkg/asset/ignition"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+)
+
+// ForIPv6Addresses creates the MachineConfig to tell kernel to configure the IP addresses with DHCP and DHCPV6.
```

Comment:
Just a side note, feel free to discard if not relevant - if you enable DHCPv6, make sure nodes don't get IPv6 address using SLAAC. We are noticing lots of issues for setups where you have both SLAAC and DHCPv6 and in principle we call such a configuration as unsupported

Reply:
@mkowalski Which issues are you seeing?
I tried this configuration multiple times on an install that was using slaac Subnet and didn't see any issues.

Is slaac supported with baremetal? I see baremetal configures dhcpv6 by default:
https://github.com/openshift/cluster-baremetal-operator/blob/master/provisioning/provisioning_info.go#L28-L29

Reply:
Have you tried dual-stack VIPs with hosts that have DHCPv6 and SLAAC at the same time? I am getting numerous reports that in this case the 2nd VIP (from the IPv6 subnet) is missing. After some investigations it turns out that if a host has multiple IPv6 addresses from the Machine Network, undesired stuff happens

Reply:
Is your questions about if we have tried dual-stack installs where the OpenShift nodes have more than one IPv6 subnet for the main interface (the one node addresses are derived from)? or is it about the main interface having one v4 and v6 and an additional interface having another v6 type?

For the first case we don't support that, in the installer we validate that only one v4 and v6 exists.
For the second case we didn't have any issues.

Reply:
2nd case is not a problem indeed.

>OpenShift nodes have more than one IPv6 subnet for the main interface (the one node addresses are derived from)?

Yes, this, but it's not about a content of install-config.yaml because this indeed you validate in o/installer; it's about output of `ip a`. Look at the example below

```
$ ip -6 a
2: wlp0s20f3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 state UP qlen 1000
    inet6 fde4:425:6b66:a::628/128 scope global dynamic noprefixroute 
       valid_lft 42524sec preferred_lft 42524sec
    inet6 2001:1711:fa41:6a0a::628/128 scope global dynamic noprefixroute 
       valid_lft 42524sec preferred_lft 42524sec
    inet6 2001:1711:fa41:6a0a:15f4:a897:a3c0:e009/64 scope global noprefixroute 
       valid_lft forever preferred_lft forever
    inet6 fde4:425:6b66:a:a848:758a:a664:6deb/64 scope global noprefixroute 
       valid_lft forever preferred_lft forever
    inet6 fe80::84ca:4a24:ffbf:1778/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever
```

`2001:1711:fa41:6a0a::628` comes from DHCPv6 and `2001:1711:fa41:6a0a:15f4:a897:a3c0:e009` comes from SLAAC. In install-config.yaml the v6 subnet is `2001:1711:fa41:6a0a::/64`. Such an attempt will fail because interface `wlp0s20f3` has 2 IP addresses that belong to the same subnet.

I am almost certain that o/installer doesn't have a validation for this because it's not something you see before you boot your node and let NetworkManager to settle

Reply:
I just tested one install with slaac and I see two addresses, but both came from the same slaac subnet. One correspond to the primary address assigned to the VM and the other correspond to the API VIP. So, no issues seen.

```
[core@ostest-jphwr-master-2 ~]$ ip a show br-ex |grep fd2
    inet6 fd2e:6f44:5dd8:c956:f816:3eff:fe78:cf36/128 scope global nodad deprecated 
    inet6 fd2e:6f44:5dd8:c956:f816:3eff:fe5e:9d93/64 scope global noprefixroute
```

In your case, could it be that users have been adding VIPs to the install-config with address type different from  the primary subnet of the nodes?

Reply:
It's hard to say as whenever I fixed some part, we were discovering yet-another-edge-case that was leading to issues. Overall the conclusion was "the less IP addresses you have on a single interface, the better".

Having said all this, I don't expect that you change anything on your side. Just that once you start using it and spot some weird things happening, remember that we had them and can probably skip some initial debugging

Reply:
ack, thanks for bringing this to my attention!